rfcs/text/0004-session.md

```diff
@@ -1,79 +1,29 @@
 # 0004: Session
 <!--^ The ECS team will assign a unique, contiguous RFC number upon merging the initial stage of this RFC, taking care not to conflict with other RFCs.-->
 
-- Stage: **0 (strawperson)** <!-- Update to reflect target stage -->
-- Date: 7/30/2020 <!-- Update to reflect date of most recent stage advancement -->
+- Stage: **2 (draft)** <!-- Update to reflect target stage -->
+- Date: TBD <!-- Update to reflect date of most recent stage advancement -->
 
-<!--
-Stage 0: Provide a high level summary of the premise of these changes. Briefly describe the nature, purpose, and impact of the changes. ~2-5 sentences.
--->
+This RFC calls for the addition of session fields to describe events related to
+various types of "sessions" reported by appliances, security devices, systems,
+management portals, applications, etc.
+
+In addition to these fields, a new `event.category` value of "session" should be added.
+Any event that captures information about a session should include "session" in
+the array field `event.category`.
 
 ## Fields
-This RFC calls for the addition of session fields to describe events related to various types of "sessions" reported by appliances, security devices, systems, management portals, applications, etc.
 
 | Field | Description |
 | ----- | ----------- |
-|session.kind:            | local, remote, network
-|session.authorization:   | user, admin, service
-|session.type:            | system, virtual, application, wired, wireless, vpn
+|session.kind:            | system, application, network
+|session.type:            | [ local, remote, virtual, wired, wireless, vpn ]
+|session.authorization:   | user, admin, service, access
```

Comment:
This compact table is a great way to visualize all of these values together.

I have a few questions:

The value "access" appears to have been dropped out of the values mentioned in the YAML description. Was this intentional? If so, please remove it from here. If it was not intentional, please add it back in the new YAML file at `rfcs/text/0004/session.yml`

I initially thought there was an overlap with values `session.kind: network` and `session.type: remote`. When adjusting the YAML file for readability it occurred to me that `session.kind: network` was likely more to identify network level sessions, rather than *anything that happens over a network*. In other words, a web cookie-based session would not use `session.kind: network`, but `session.kind: application`. Am I understanding this correctly?

However within the different values for `session.type`, I do wonder if there's overlap. It seems like "wired", "wireless" and "vpn" would always be accompanied by value "remote". Is this also how you see this? Note that this is not necessarily a problem. "local" vs "remote" could be useful when plotted against one another, whereas "wired", "wireless" & "vpn" could be useful to break down the kinds of "remote" session types.

One final overlap concern: I do think `session.kind: system` and `session.authorization: service` are almost entirely overlapping, aren't they? Should we get rid of one of them, or am I misunderstanding a subtlety?

Reply:
> One final overlap concern: I do think session.kind: system and session.authorization: service are almost entirely overlapping, aren't they? Should we get rid of one of them, or am I misunderstanding a subtlety?

I was trying to find discrete combinations that would uniquely identify <user> logging into a system, and a service running with authentication/authorization included in the "session".  It may be unnecessary (I'll look thru auditbeat service logs and see what they look like...  could just be that a system service is not something that would typically be defined as a session (e.g. starting up a vpn connection on a pfsense linux firewall - presumably the vpn application logs would be sufficient)

> The value "access" appears to have been dropped...

the revamp of the fields seems to lend itself more towards the entity being authorized - authentication would be a part of the overall session, but I don't think it makes sense as a specific session field.  Basically the split becomes: 

- "user interactive session" (user logs in or connects to a service or application to use said service/app)
- "admin interactive session"(user logs in to manage a service or application e.g. root, enable mode in cisco parlance) 
- "Network session" (user or system connects to a monitored network)

Not all of the type values will apply to every session.kind:
Session Kind:
- System (full system access)
  - local (hands on keyboard)
  - remote (bash via ssh)
  - virtual (Citrix -virtual desktop)
- Application
  - remote (e.g. web connection, smb share, ftp)
  - local (possible but I can't think of an example)
  - virtual (citrix published apps)
-Network
  - wired (802.1x network session - e.g. time connected to the LAN)
  - wireless (802.1x network session - e.g. time connected to the WLAN)
  - vpn (user connects to vpn concentrator

I think this one could use some more input- I can see session.type having fixed single values (virtual implies remote), or using an array (vpn would be network & remote, published apps & citrix desktop would be virtual & remote, opening a virtualbox session locally could be local & virtual...)

Does that all make sense?

Wired & wireless however would typically not use remote as the session is for the local user or workstation directly connecting to the local network (as opposed to connecting to another entity over the network)

Reply:
I'm actually wondering if virtual needs to be specifically identified... a "remote" "system" session will be telnet, ssh, rdp, or citrix... not sure the virtual type is necessary to split out rdp/citrix

Reply:
And service as an authorization could also make sense for workstation based network access (802.1x cert auth), network to network vpn, etc...